src/lib/graph/build-graph.ts

```diff
@@ -47,20 +59,21 @@ export class BuildGraph implements Traversable<Node> {
     return Array.from(values);
   }
 
-  public some(by: (value: Node, index: number) => boolean): boolean {
+  public some(by: SimplePredicate): boolean {
     return this.entries().some(by);
   }
 
-  public filter(by: (value: Node, index: number) => boolean): Node[] {
-    return this.entries().filter(by);
+  public filter<T extends Node = Node>(by: ComplexPredicate<Node, T>): T[] {
+    return this.entries().filter(by) as T[];
   }
 
-  public find(by: (value: Node, index: number) => boolean): Node | undefined {
-    return this.entries().find(by);
+  public find<T extends Node = Node>(by: ComplexPredicate<Node, T>): T | undefined {
+    return this.entries().find(by) as T;
   }
 
   public from(node: Node): Traversable<Node> {
-    return node;
+    // TODO: Remove or fix this method.
+    return (node as unknown) as Traversable<Node>;
```

Comment:
This method isn't used. Should we remove it?

Reply:
Yes please